src/ppc64.h

```diff
@@ -45,18 +45,40 @@
 #define DES_COPY			0
 #define DES_BS_ASM			0
 #define DES_BS				1
-#define DES_BS_VECTOR			0
+#if defined(JOHN_ALTIVEC)
+#define DES_BS_EXPAND			3
+#if 1
+#define DES_BS_VECTOR			2
+#define DES_BS_ALGORITHM_NAME		"DES 128/128 AltiVec"
+#elif 0
+/* It is likely unreasonable to use S-box expressions requiring vsel when this
+ * operation is only available in one of the two instruction sets.
+ * So let's revert to less demanding S-box expressions. */
+#define DES_BS_VECTOR			3
+#define DES_BS_VECTOR_SIZE		4
+#define DES_BS_ALGORITHM_NAME		"DES 128/128 AltiVec + 64/64"
+#else
+#define DES_BS_VECTOR			4
+#define DES_BS_ALGORITHM_NAME		"DES 128/128 X2 AltiVec"
+#endif
+#else
 #define DES_BS_EXPAND			1
+#define DES_BS_VECTOR			0
+#endif
 
-#define MD5_ASM				0
+#define MD5_ASM			0
 #define MD5_X2				1
-#define MD5_IMM				0
+#if HAVE_ALTIVEC
+#define MD5_IMM			1
+#else
+#define MD5_IMM			0
+#endif
```

Comment:
What version of `ppc64.h` (or what else?) did you start with? I have recently revised it in core to set `MD5_IMM` to 1 even for non-AltiVec, as that proved faster now. I mentioned this detail on a GitHub issue a couple of weeks ago.